src/foraging_gui/RigJsonBuilder.py

```diff
@@ -0,0 +1,681 @@
+    ###########################################################################
+    components['mouse_platform']=d.Tube(
+        name="mouse_tube_foraging", 
+        diameter=3.0,
+        diameter_unit=SizeUnit.CM
+        )
+
+
+    # Stimulus devices
+    ###########################################################################
+    if settings['newscale_serial_num_box{}'.format(settings['box_number'])] != '':
+        stage = d.MotorizedStage(
+                    name="NewScaleMotor for LickSpouts",
+                    serial_number=settings['newscale_serial_num_box{}'.format(settings['box_number'])], 
+                    manufacturer=d.Organization.NEW_SCALE_TECHNOLOGIES,
+                    travel=15.0,
+                    travel_unit=SizeUnit.MM,
+                    firmware="https://github.com/AllenNeuralDynamics/python-newscale, branch: axes-on-target, commit #7c17497",
+                    )
+    else:
+        stage = d.MotorizedStage(
+                    name="AIND lick spout stage",
+                    manufacturer=d.Organization.AIND,
+                    travel=30, 
+                    travel_unit=SizeUnit.MM
+                    )      
+
+    if ('AINDLickDetector' in settings['box_settings']) and (settings['box_settings']['AINDLickDetector'] == "1"):
+        lick_spout_name = 'AIND_Lick_Detector'
+    else:
+        lick_spout_name = 'Janelia_Lick_Detector'
+    lick_spouts=[
+        d.RewardSpout(
+            name="{} Left".format(lick_spout_name),
+            side=d.SpoutSide.LEFT,
+            spout_diameter=1.2,
+            solenoid_valve=d.Device(device_type="Solenoid", name="Solenoid Left"),
+            lick_sensor_type=d.LickSensorType("Capacitive")
+        ),
+        d.RewardSpout(
+            name="{} Right".format(lick_spout_name),
+            side=d.SpoutSide.RIGHT,
+            spout_diameter=1.2,
+            spout_diameter_unit=SizeUnit.MM,
+            solenoid_valve=d.Device(device_type="Solenoid", name="Solenoid Right"),
+            lick_sensor_type=d.LickSensorType("Capacitive")
+        ),
+        ]   
+
+    components['stimulus_devices']=[
+        d.RewardDelivery(
+            reward_spouts=lick_spouts,
+            stage_type = stage,
+        ),
+        d.Speaker(
+            name="Stimulus speaker",
+            manufacturer=d.Organization.OTHER,
+        )
+        ]
+
+
+    # DAQS
+    ###########################################################################
+
+    components['daqs']=[
+        d.HarpDevice(
+            name="Harp Behavior",
+            harp_device_type=d.HarpDeviceType.BEHAVIOR,
+            manufacturer=d.Organization.CHAMPALIMAUD,
+            core_version="2.1",
+            firmware_version="FTDI version:",
+            computer_name=settings['computer_name'], 
+            is_clock_generator=False,
+            channels=[
+                d.DAQChannel(channel_name="DO0", device_name="Solenoid Left", channel_type="Digital Output"),
```

Comment:
This is not correct. Both Solenoid and lick spouts are both via Ethernet cable.

Reply:
I think @hagikent wrote this part. What is the correct way to document how the solenoid and lick spouts are connected? I could write `data_interface=d.DataInterface.ETH` which corrects the issue you mentioned, but then its not clear how the solenoids and lick spouts are connected to the behavior board

Reply:
I remember I wrote this with Saskia's input during the Ephys-turned-metadata Hackathon.
Why Ethernet cables need to be documented? The relevant info would be port/channel number?

Reply:
DO0 and DO1 are other channels in the behavior board, not from the Ethernet cables.

Reply:
Yes, Xinxin is correct. DO0 and DO1 are separate ports from the ethernet ports.

How about:

`
   d.HarpDevice(
            name="Harp Behavior",
            harp_device_type=d.HarpDeviceType.BEHAVIOR,
            manufacturer=d.Organization.CHAMPALIMAUD,
            computer_name=settings['computer_name'], 
            is_clock_generator=False,
            data_interface = d.DataInterface.ETH,
            notes="Lick detectors, and reward delivery solenoids are connected via ethernet cables"
            )
 `

Reply:
yeah, looks good.

Reply:
Updated

Reply:
I don't know what's the official name of those Eth ports but can we make it consistent with how we call it in software? In bonsai, they are called `DigitalOutput/SupplyPort0(or1)` and Saskia and I decided to define DO0or1.
Otherwise super confusing especially when making the updated diagram.

![Screenshot 2024-05-22 at 16 00 23](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/assets/54397345/f2670323-4087-42e6-9386-249ec40666b8)

Reply:
And to be a bit more informative for @alexpiet , two ethernet cables are, in parallel, connecting HarpBehavior and Janelia-lick-detector(+solenoid activator). 0 and 1 are corresponding to L and R solenoids, respectively.

Reply:
I dont have any idea what we should call it. All I know is that DO0 and DO1 are physically marked on the boards at a different port

Reply:
>DO0, DI1, 

I see, seems yes I confirmed in the below link....

According to this, hardware-wise they are apparently called `Poke` as intended usage is to connect NosePoke device?
https://github.com/harp-tech/device.behavior

@XX-Yin 
Why at the software level they are SupplyPort0? Shouldn't be matching? Too confusing.

Reply:
This is real.  Different naming can be confusing.

 I think the `Poke` is more related to the tasks they originally design for. They are multiple connections integrated in the Ethernet connector, and `SupplyPort0` is more related to the general function, from one of the ethernet connections, to `give 12v voltage output`.

Reply:
> This is real. Different naming can be confusing.
> 
> I think the `Poke` is more related to the tasks they are originally designed for. They are multiple connections integrated in the Ethernet connector, and `SupplyPort0` is more related to the general function, from one of the ethernet connections, to `give 12v voltage output`.